main.cpp

```diff
@@ -435,13 +444,31 @@ void Mainloop::loop()
                 handle_tcp_connection();
                 continue;
             }
-            Endpoint *e = static_cast<Endpoint*>(events[i].data.ptr);
+            Pollable *p = static_cast<Pollable *>(events[i].data.ptr);
+
+            if (Endpoint *e = dynamic_cast<Endpoint *>(p)) {
+                if (events[i].events & EPOLLIN)
+                    handle_read(e);
+
+                if (events[i].events & EPOLLOUT)
+                    handle_canwrite(e);
+
+                continue;
+            }
+
+            if (Timeout *t = dynamic_cast<Timeout *>(p)) {
+                uint64_t val = 0;
```

Comment:
So add this methods to Pollable:
```
virtual int handle_read(struct buffer *pbuf) = 0;
virtual int handle_write(const struct buffer *pbuf) = 0;
```
And rename this methods of Endpoint to those methods above?
```
int read_msg(struct buffer *pbuf);
virtual int write_msg(const struct buffer *pbuf) = 0;
```
And implement handle_read() of Timeout to handle everything intern and always return 0?